code/modules/projectiles/guns/ballistic/launchers.dm

```diff
@@ -89,8 +89,10 @@
 		This one has been fitted with a special backblast diverter to prevent 'friendly' fire 'accidents' during use."
 	backblast = FALSE
 
-/obj/item/gun/ballistic/rocketlauncher/afterattack()
+/obj/item/gun/ballistic/rocketlauncher/try_fire_gun(atom/target, mob/living/user, params)
 	. = ..()
+	if(!.)
+		return
 	magazine.get_round(FALSE) //Hack to clear the mag after it's fired
```

Comment:
should we move clearing the magazine to invoke on an override of `fire_gun()` e.g. `type/fire_gun(...); . = ..(); magazine.get_round(FALSE)`? i don't really know how i feel about using try_fire_gun() like this since everything else seems to use it as a checker proc to invoke parent, not work after parent. feels awk but up to you.

Reply:
that's fine then